src/building/suggested.md

```diff
@@ -49,22 +49,62 @@ you can write: <!-- date-check: nov 2022 --><!-- the date comment is for the edi
         "--json-output"
     ],
     "rust-analyzer.cargo.sysroot": "./build/host/stage0-sysroot",
-    "rust-analyzer.rustc.source": "./Cargo.toml",
+    "rust-analyzer.rustc.source": "./Cargo.toml"
 }
 ```
 
 in your `.vscode/settings.json` file. This will ask `rust-analyzer` to use
 `./x.py check` to check the sources, and the stage 0 rustfmt to format them.
 
-If you have enough free disk space and you would like to be able to run `x.py` commands while
-rust-analyzer runs in the background, you can also add `--build-dir build-rust-analyzer` to the
-`overrideCommand` to avoid x.py locking.
+There are several ways for configuring neovim for rustc. The easiest way is to use the plugin 
+[neoconf.nvim](https://github.com/folke/neoconf.nvim/), which is compatible with VSCode (and 
+other) config formats. To manage configuration files use the `:Neoconf local` command.
 
 If you're running `coc.nvim`, you can use `:CocLocalConfig` to create a
 `.vim/coc-settings.json` and enter the same settings as above, but replacing
 `editor.formatOnSave: true,` with
 `"coc.preferences.formatOnSaveFiletypes": ["rust"],`.
 
+Another way is without a plugin, and creating your own logic in your configuration. The required 
+Lua for doing so is below.
+
+```lua
+{
+    ["rust-analyzer"] = {
+        checkOnSave = {
+            overrideCommand = {
+                {"python3", "x.py", "check", "--json-output"}
+            }
+        },
+        rustfmt = {
+            overrideCommand = {
+                {"./build/host/stage0/bin/rustfmt", "--edition=2021"}
+            }
+        },
+        procMacro = {
+            server = "./build/host/stage0/libexec/rust-analyzer-proc-macro-srv",
+            enable = true
+        },
+        cargo = {
+            buildScripts = {
+                enable = true,
+                invocationLocation = "root",
+                invocationStrategy = "once",
+                overrideCommand = {"python3", "x.py", "check", "--json-output"}
+            },
+            sysroot = "./build/host/stage0-sysroot"
+        },
+        rustc = {
+            source = "./Cargo.toml"
+        }
+    }
+}
```

Comment:
I'd avoid adding all this again because we already have a json file in the rust repo.

Reply:
Is that in there? That is the Lua config, not the JSON config. We could also put the Lua config in the Rust repo, but there's not really a good way for `x config`  to automatically configure it.

Reply:
Could just delete it and leave only one "suggested" way to configure it. This is for people who don't want to use an extension that can read the JSON file.

Reply:
Could just delete it and leave only one "suggested" way to configure it. This is for people who don't want to use an extension that can read the JSON file.

Reply:
Yep, I got the intention of adding this. I'd try to suggest some docs about how to convert from the json to Lua config. I'm worried that having this duplicated information would lead us to have places outdated. From what I've been seeing RA config is changing frequenly so my fear is that things are not in sync and outdated or either in the Lua one or in the json one.

cc @jyn514

Reply:
Yeah I have the same worry. Especially since they aren't living in the same document anymore. I'd like to make it easier for people who don't want to use plugins, but also those people probably know how to translate the JSON in to Lua? But I don't really know.

Reply:
Sounds good, I'll do that when I get home.